pkg/muxer/tcp/mux.go

```diff
@@ -315,7 +315,7 @@ func alpn(tree *matchersTree, protos ...string) error {
 	return nil
 }
 
-var almostFQDN = regexp.MustCompile(`^[[:alnum:]\.-]+$`)
+var almostFQDN = regexp.MustCompile(`^[[:alnum:]\.\-\:]+$`)
```

Comment:
```suggestion
var almostFQDN = regexp.MustCompile(`^[[:alnum:]\.\:-]+$`)
```

Reply:
the suggestion is not right: `-` must be escaped.
If you want to remove non require escape: `^[[:alnum:].\-:]+$`